testing/scenario_app/ios/Scenarios/ScenariosUITests/AppExtensionTests.m

```diff
@@ -27,13 +32,25 @@ - (void)testAppExtensionLaunching {
   }
   [button tap];
   BOOL launchedExtensionInFlutter = NO;
-  // Custom share extension button (like the one in this test) does not have a unique
-  // identity. They are all identified as `XCElementSnapshotPrivilegedValuePlaceholder`.
-  // Loop through all the `XCElementSnapshotPrivilegedValuePlaceholder` and find the Flutter one.
+
+  // Wait for first cell of share sheet to appear.
+  XCUIElement* firstCell = self.hostApplication.collectionViews.cells.firstMatch;
+  if (![firstCell waitForExistenceWithTimeout:10]) {
+    NSLog(@"%@", self.hostApplication.debugDescription);
+    XCTFail(@"Failed due to not able to find any cells with %@ seconds", @(10));
+  }
```

Comment:
On Xcode 15/iOS 17, sometimes it would process `self.hostApplication.collectionViews.cells.count` below before the sheet appeared and therefore would be 0. So I added this to wait for it to populate.